alf/environments/metadrive/geometry.py

```diff
@@ -163,6 +193,68 @@ def batched(self) -> bool:
         """
         return self.point.ndim == 3
 
+    def transformed(self, center: np.ndarray, orientation: float) -> Polyline:
+        """Returns a transformed (set of) polyline(s) so that the resulting points are
+        in the coordinate frame defined by the pose (center and the orientation).
+
+        Args:
+
+            center: A 2D point denoting the origin of the new coordinate frame.
+            orientation: orientation (x-axis direction) of the new coordinate in radian.
+
+        """
+        # Construct the 2D rotation matrix.
+        cos = np.cos(orientation)
+        sin = np.sin(orientation)
+        rotation = np.array([[cos, -sin], [sin, cos]])
+
+        if self.batched:
+            transformed = np.einsum('ijk,kl->ijl', (self.point - center),
+                                    rotation)
```

Comment:
this branch is unnecessary. matmul can handle both cases.

Reply:
Yes you are right. For some reason I mistakenly remembered pytorch allows this but numpy does not ...